src/models/ConstrainedMetaModel.ts

```diff
@@ -2,16 +2,23 @@ import { makeUnique } from '../helpers/DependencyHelpers';
 import {
   MetaModel,
   MetaModelOptions,
-  MetaModelOptionsConst
+  MetaModelOptionsConst,
+  MetaModelOptionsDiscriminator
 } from './MetaModel';
 
 export interface ConstrainedMetaModelOptionsConst
   extends MetaModelOptionsConst {
   value?: unknown;
 }
 
+export interface ConstrainedMetaModelOptionsDiscriminator
+  extends MetaModelOptionsDiscriminator {
+  type?: string;
+}
+
 export interface ConstrainedMetaModelOptions extends MetaModelOptions {
   const?: ConstrainedMetaModelOptionsConst;
+  discriminator?: ConstrainedMetaModelOptionsDiscriminator;
```

Comment:
Dont think we need to constrain discriminator, as the value will always be the same. I.e. it does not change from language to language.

Reply:
As you can see in `addDiscriminatorTypeToUnionModel`, I want to calculate if the discriminator property has a common type. I will use this property in the Union Model (interface) for Java if it exists. I can move this logic to the Java Union Renderer, but this seems language-agnostic to me.

Reply:
Hmmm, I dont think I fully understand how you are gonna use it, because you are taking a different direction then I would 😄 

I would have added a discriminator value to each model in the union, that the serializer can iterate to check which one is the appropriate model to pick.

But lets just go forward with your approach, I am sure it falls into place when the generators change 😄